src/commands/edge/drivers/logcat.ts

```diff
@@ -147,27 +156,47 @@ export default class LogCatCommand extends SseCommand {
 		const liveLogPort = port ?? DEFAULT_LIVE_LOG_PORT
 		this.authority = `${ipv4}:${liveLogPort}`
 
-		this.logClient = new LiveLogClient(this.authority, this.authenticator, this.checkServerIdentity.bind(this))
+		const config: LiveLogClientConfig = {
+			authority: this.authority,
+			authenticator: this.authenticator,
+			verifier: this.checkServerIdentity.bind(this),
+			timeout: flags['connect-timeout'],
+		}
+
+		this.logClient = new LiveLogClient(config)
 	}
 
 	async run(): Promise<void> {
-		const installedDriversPromise = this.logClient.getDrivers()
+		CliUx.ux.action.start('connecting')
+
+		// ensure host verification resolves before connecting to the event source
+		const installedDrivers = await this.logClient.getDrivers()
 
 		let sourceURL: string
 		if (this.flags.all) {
 			sourceURL = this.logClient.getLogSource()
 		} else {
-			const driverId = await this.chooseHubDrivers(this.args.driverId, installedDriversPromise)
+			const driverId = await CliUx.ux.action.pauseAsync(() => this.chooseHubDrivers(this.args.driverId, installedDrivers))
 			sourceURL = driverId == DEFAULT_ALL_TEXT ? this.logClient.getLogSource() : this.logClient.getLogSource(driverId)
 		}
 
-		// ensure this resolves before connecting to the event source
-		const installedDrivers = await installedDriversPromise
-
-		CliUx.ux.action.start('connecting')
 		await this.initSource(sourceURL)
 
+		const sourceTimeoutID = setTimeout(() => {
+			this.teardown()
+			CliUx.ux.action.stop('failed')
+			try {
+				handleConnectionErrors(this.authority, 'ETIMEDOUT')
+			} catch (error) {
+				if (error instanceof Error) {
+					Errors.handle(error)
+				}
+			}
+		}, this.flags['connect-timeout']).unref() // unref lets Node exit before callback is invoked
+
```

Comment:
Setting timeout in this fashion due to https://github.com/EventSource/eventsource/issues/251.